host/class/uac/usb_host_uac/include/usb/uac_host.h

```diff
@@ -0,0 +1,393 @@
+ * - ESP_ERR_INVALID_ARG if the device handle or stream configuration is invalid
+ * - ESP_ERR_NOT_FOUND if the stream configuration is not supported
+ * - ESP_ERR_INVALID_STATE if the device is not in the right state
+ * - ESP_ERR_NO_MEM if memory allocation failed
+ * - ESP_ERR_TIMEOUT if the control transfer timeout
+ */
+esp_err_t uac_host_device_start(uac_host_device_handle_t uac_dev_handle, const uac_host_stream_config_t *stream_config);
+
+/**
+ * @brief Suspend a UAC stream
+ *
+ * @param[in] uac_dev_handle  UAC device handle
+ * @return esp_err_t
+ * - ESP_OK on success
+ * - ESP_ERR_INVALID_ARG if the device handle is invalid
+ * - ESP_ERR_INVALID_STATE if the device is not in the right state
+ */
+esp_err_t uac_host_device_suspend(uac_host_device_handle_t uac_dev_handle);
+
+/**
+ * @brief Resume a UAC stream with same stream configuration
+ *
+ * @param[in] uac_dev_handle  UAC device handle
+ * @return esp_err_t
+ * - ESP_OK on success
+ * - ESP_ERR_INVALID_ARG if the device handle is invalid
+ * - ESP_ERR_INVALID_STATE if the device is not in the right state
+ */
+esp_err_t uac_host_device_resume(uac_host_device_handle_t uac_dev_handle);
+
+/**
+ * @brief Stop a UAC stream, stream resources will be released
+ *
+ * @param[in] uac_dev_handle  UAC device handle
+ * @return esp_err_t
+ * - ESP_OK on success
+ * - ESP_ERR_INVALID_ARG if the device handle is invalid
+ * - ESP_ERR_INVALID_STATE if the device is not in the right state
+ */
+esp_err_t uac_host_device_stop(uac_host_device_handle_t uac_dev_handle);
+
+/**
+ * @brief Read data from UAC stream
+ *
+ * @param[in] uac_dev_handle  UAC device handle
+ * @param[out] data           Pointer to the buffer to store the data
+ * @param[in] size            Number of bytes to read
+ * @param[out] bytes_read     Pointer to the number of bytes read
+ * @param[in] timeout         Timeout in ticks. For milliseconds, please use 'pdMS_TO_TICKS()' macros
+ * @return esp_err_t
+ * - ESP_OK on success
+ * - ESP_ERR_INVALID_ARG if the device handle or data is invalid
+ * - ESP_ERR_INVALID_STATE if the device is not in the right state
+ */
+esp_err_t uac_host_device_read(uac_host_device_handle_t uac_dev_handle, uint8_t *data, uint32_t size,
+                               uint32_t *bytes_read, uint32_t timeout);
+
+/**
+ * @brief Write data to UAC stream
+ *
+ * @param[in] uac_dev_handle  UAC device handle
+ * @param[in] data            Pointer to the data buffer
+ * @param[in] size            Number of bytes to write
+ * @param[in] timeout         Timeout in ticks. For milliseconds, please use 'pdMS_TO_TICKS()' macros
+ * @return esp_err_t
+ * - ESP_OK on success
+ * - ESP_ERR_INVALID_ARG if the device handle or data is invalid
+ * - ESP_ERR_INVALID_STATE if the device is not in the right state
+ * - ESP_FAIL if write failed or timeout
+*/
+esp_err_t uac_host_device_write(uac_host_device_handle_t uac_dev_handle, uint8_t *data, uint32_t size,
+                                uint32_t timeout);
```

Comment:
Docs: Could you please add a note that would say that these functions return _before_ the USB transfer is complete?

Because, in all other USB Host drivers the read/write operations also take `timeout` parameter, but they return only _after_ the transfer is completed. In UAC we return after that data is added to the ringuf

Reply:
Agree!